frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -168,6 +168,46 @@ function VirtualKey:init()
             face = self.face,
             bold = self.bold or false,
         }
+    -- make long labeled keys to have a smaller font (c.f. https://github.com/koreader/koreader/pull/7972#issuecomment-884822411)
+        local max_width = self.width - 2*self.bordersize - 2*Size.padding.small
+        while label_widget:getWidth() > max_width do
+            local new_size = label_widget.face.orig_size - 1
+            label_widget:free()
+            if new_size < 8 then break end -- don't go too small
+            label_widget = TextWidget:new{
+                text = self.label,
+                face = Font:getFace(self.face.orig_font, new_size),
+                bold = self.bold or false,
+            }
+        end
+    end
+    --if not self.alt_label then self.alt_label = "z" end -- for testing
```

Comment:
Remove this, it was just for our initial tests.

Reply:
OK!